### ISSUE TITLE
Private methods, but and does

### ISSUE THREAD

**Issue by JJ** [Raku/doc#3523]
## Problem or new feature

Apparently, neither but nor does are able to access private attributes or methods:

```perl6
role Privateer {
    method test-private-method ( $method-name, |c  ) {
	self!"$method-name"(|c);
    }
}

class Privateed does Privateer {
    method !private() { return "⌣"  }
}

class Random {
    method !private() { return "⛔" }
}
my $obj = Privateed.new;
say $obj.test-private-method( "private" );

$obj = Random.new;
$obj does Privateer;
say $obj.test-private-method( "private" );

$obj = Random.new but Privateer;
say $obj.test-private-method( "private" );
```

The error here will be along the lines of 
```
No such private method '!private' for invocant of type 'Random+{Privateer}'
```

## Suggestions

Investigate this, and also if there's any difference between `but` and `does`.

**Comment by jnthn** [comment]
The `does` and `but` operators perform a mixin, which means we:

0. If it's `but`, make a clone and work on that in the following steps instead.
1. Create an anonymous subclass
2. Make it inherit from the current type of the object
3. Compose the role into it
4. Switch the type of the object to the subclass

Since the role is composed into a subclass, then it's not in `Random`, and so can't see its private methods.

**Comment by JJ** [comment]
Yes, that's implicit in the description, but it should probably be clearer.
